src/main/java/com/ita/if103java/ims/dao/impl/EventDaoImpl.java

```diff
@@ -58,104 +59,40 @@ public Event findById(Long id) {
-        }
-    }
-
-    @Override
-    public List<Event> findByWarehouseIdAndType(Long warehouseId, EventType type) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_TYPE, eventRowMapper, warehouseId, type.toString());
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
-        }
-    }
-
-    @Override
-    public List<Event> findByWarehouseIdAndDate(Long warehouseId, ZonedDateTime date) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_DATE, eventRowMapper, warehouseId, date.toLocalDate().toString());
+            throw new EventNotFoundException("Failed to obtain event during `select` {" + where + "}, EventDao.findAll", e);
         } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+            throw new CRUDException("Error during `select` {" + where + "}, EventDao.findAll", e);
         }
     }
 
-    @Override
-    public List<Event> findByAccountIdAndDate(Long accountId, ZonedDateTime date) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_ACCOUNT_ID_AND_DATE, eventRowMapper, accountId, date.toLocalDate().toString());
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+    private String buildSqlFilterCondition(String columnName, Object columnValue) {
+        if (columnName == "date") {
+            return String.format("DATE(%s) %s '%s'", columnName, columnValue == null ? "is" : "=", columnValue);
         }
-    }
-
-    @Override
-    public List<Event> findByWarehouseIdAndWithinDates(Long warehouseId, ZonedDateTime after, ZonedDateTime before) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_WAREHOUSE_ID_AND_WITHIN_DATES, eventRowMapper,
-                warehouseId, after.toLocalDate().toString(), before.toLocalDate().toString());
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+        if (columnName == "after") {
+            return String.format("DATE(date) %s '%s'", columnValue == null ? "is" : ">=", columnValue);
         }
-    }
-
-    @Override
-    public List<Event> findByAccountIdAndWithinDates(Long accountId, ZonedDateTime after, ZonedDateTime before) {
-        try {
-            return jdbcTemplate.query(Queries.SQL_SELECT_EVENTS_BY_ACCOUNT_ID_AND_WITHIN_DATES, eventRowMapper, accountId,
-                after.toLocalDate().toString(), before.toLocalDate().toString());
-        } catch (DataAccessException e) {
-            throw crudException(e.getMessage(), "get", "*");
+        if (columnName == "before") {
+            return String.format("DATE(date) %s '%s'", columnValue == null ? "is" : "<=", columnValue);
         }
+        return String.format("%s %s '%s'", columnName, columnValue == null ? "is" : "=", columnValue);
```

Comment:
I would skip null values rather than building condition for them.

But if from UI you are going to not send keys for which you do not want to filter then your approach can work.

However your approach doesn't handle List of items as value

Reply:
let's discuss this today, please remind me